src/forms/inputcounter/InputCounter.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+import * as React from "react";
+import * as utils from "../../utils";
+import {Icon} from "../../icon";
+import {Input} from "../input/Input";
+import "./InputCounter.css";
+
+type InputCounterProps = {
+    className?: string;
+    maxValue?: number;
+    minValue?: number;
+    defaultValue?: number;
+    errMsg?: string;
+    width?: string;
+    onChange?: (evt: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+type InputCounterState = {
+    value: number;
+    errorMsg: string;
+};
+
+export class InputCounter extends React.PureComponent<InputCounterProps, InputCounterState> {
+    public static defaultProps: Partial<InputCounterProps> = {
```

Comment:
you can remove public keyword here. Also not getting what is use of 'Partial' here ?

Reply:
This "Partial<Props>" is used to assign value to partial props as default value in typescript version < 3. As we are using typescript version > 3, we can remove that.